alacritty/src/display/hint.rs

```diff
@@ -223,6 +223,25 @@ impl HintMatch {
     }
 }
 
+/// A hint being highlighted by the mouse or vi cursor.
+#[derive(PartialEq, Eq)]
+pub struct HighlightedHint {
+    pub last_cell_id: u32,
+    pub hint: HintMatch,
+}
+
+impl HighlightedHint {
+    fn new(hint: HintMatch, last_cell_id: u32) -> Self {
+        Self { last_cell_id, hint }
+    }
+
+    /// Check if the bounds of the highlight are still valid.
+    pub fn has_valid_range<T>(&self, term: &mut Term<T>) -> bool {
+        let current_id = term.grid_mut()[*self.hint.bounds.end()].id();
+        self.last_cell_id == current_id
+    }
```

Comment:
Why does this need to take a `&mut Term` instead of just a `&Term`. I'm sure this isn't a mistake, I'm just not sure why since all it seems you're doing here is checking an id.

Reply:
Because the `id` function dynamically assigns a new ID to the cell.